finder_sidebar_editor/cli.py

```diff
@@ -0,0 +1,155 @@
+       force: Annotated[
+           Optional[bool], typer.Option(help="ignore missing favorite name, otherwise return error.")] = False,
+       by_path: Annotated[Optional[bool], typer.Option()] = False):
+    if not name_or_path or name_or_path.isspace():
+        print_error("name or path is empty or whitespace only")
+        exit(1)
+    sidebar = FinderSidebar()
+    if by_path:
+        sidebar.remove_by_path(name_or_path)
+    else:
+        name = name_or_path  # for clarity
+        if not force:
+            sidebar.update()
+            validate_name(sidebar, name)
+            sidebar.remove(name_or_path)
+        else:
+            sidebar.remove(name_or_path)
+
+
+@app.command()
+def add(path: Annotated[str, typer.Argument(help="""The path to add, which must exist unless --force is used.
+            If you add a path that was already added, it is effectively a noop (or appears to be?).
+            The name of the favorite is the tail end of the path (i.e. /path/to/mydir gets a name mydir).
+            """)],
+        uri: Annotated[Optional[str], typer.Argument(
+            help='If this starts with "afp" or "smb", then try to mount it first. It is an error if it MacOS cannot ' \
+                 ' mount it.')] = FILE_LOCALHOST,
+        order: Annotated[Optional[int], typer.Argument(
+            help="Not yet implemented.  Added items are always at the top. Use mv to move them.")] = 0,
+        force: Annotated[Optional[bool], typer.Option(help="""
+            ignore non-existent path testing in two different ways:
+            1) if only path given, make sure that path exists in the filesystem, otherwise return error. 
+            2) if path and uri given, and uri starts with "afp" or "smb", then do the same with the url+path, otherwise error.
+            Note that forcing the add of a path that doesnt exist may lead to a damaged item displayed as '?" in your favorites in finder
+            """)] = False,
+        verbose: Annotated[Optional[bool], typer.Option()] = False
+        ):
+    sidebar = FinderSidebar()
+    if not force:
+        import os
+        from urllib.parse import urlparse
+        if not os.path.exists(path):
+            print_error(fr'"{path}" not found')
+            exit(1)
+        if not uri == FILE_LOCALHOST:
+            p = urlparse(uri + path)
+            final_path = os.path.abspath(os.path.join(p.netloc, p.path))
+            if not os.path.exists(final_path):
+                print_error(fr'"{final_path}" not found')
+                exit(1)
+    print_verbose(fr'adding "{path}" as {uri}', verbose)
+    sidebar.add(path, uri)
+
+
+@app.command()
+def mv(to_move: Annotated[str, typer.Argument()],
+       to_after: Annotated[str, typer.Argument()]):
+    sidebar = FinderSidebar()
+    sidebar.move(to_move, to_after)
+
+
+@app.command()
+def rename(from_name: Annotated[str, typer.Argument()],
+           to_name: Annotated[str, typer.Argument()]):
```

Comment:
I don't think there should be a rename command if it's not implemented.

Reply:
Hahaha, I was prodding you to make a rename method in your lib but Ill kill this.

Reply:
hahahaha, I hadn't considered it. I've looked it up and it should be feasible as `DisplayName` and `URL` are different parameters in [LSSharedFileListInsertItemURL][1]. However, if I were to fiddle with this I'd try to look up how to replace these deprecated methods. What prevents me from doing researching on this is that I don't have access to a macOS on a daily basis.

[1]: https://developer.apple.com/documentation/coreservices/1444471-lssharedfilelistinsertitemurl